fhir2dataset/url_builder.py

```diff
@@ -51,70 +51,42 @@ def _compute_url(self) -> str:
-                    # add assert search_param in CapabilityStatement
-                    key = f"{to_resource or ''}{search_param}"
-                    value = f"{values['prefix'] or ''}{values['value']}"
-                    self._params[key] = value
-
-                logger.debug(f"the part of the url for the params is: {self._params}")
-
-    # To change because it's useless to go through dijstra for the moment knowing that we only do
-    # chain parameters of length 1.
-    def _light_chained_params(self, resource_alias: str) -> tuple:
-        """gives the prefix (in the first element of the output tuple) to make a chained parameter from the main resource to the resource given as argument. If the resource given as argument is not a neighbor of the main resource, the second element of the output tuple is set to false
-
-        Arguments:
-            resource_alias {str} -- alias of a resource
 
-        Returns:
-            tuple -- (prefix, boolean)
+        The FHIR API makes union when "where conditions" are requested for neighbouring resources 
+        Only one "where condition" on every neighbouring resource is taken into account
         """  # noqa
-        to_resource = None
-        reliable = True
-        # Construction of the path from the main resource to the
-        # resource on which the parameter(s) will be applied
-        if resource_alias != self.main_resource_alias:
-            reliable = False
-        return to_resource, reliable
+        for resource_alias in self._query_graph.resources_alias_graph.neighbors(
+            self.main_resource_alias
+        ):
+            edge = self._query_graph.resources_alias_graph.edges[
+                self.main_resource_alias, resource_alias
+            ]
+
+            infos_alias = self._query_graph.resources_alias_info[resource_alias]
+            infos_search_param = infos_alias["search_parameters"]
+
+            # check if there are "where conditions" on resource_alias
+            if infos_search_param:
+                # select a random "where condition" on resource_alias
+                search_param = random.choice(list(infos_search_param))
+                values = infos_search_param[search_param]
+                searchparam_prefixe = edge[self.main_resource_alias]["searchparam_prefix"]
+
+                key = f"{searchparam_prefixe}{search_param}"
+                value = f"{values['prefix'] or ''}{values['value']}"
+                self._params[key] = value
 
-    def _chained_params(self, resource_alias: str) -> tuple:
-        """gives the prefix (in the first element of the output tuple) to make a chained parameter from the main resource to the resource given as argument. If the resource given as argument is not a neighbor of the main resource, the second element of the output tuple is set to false
+                logger.debug(f"the part of the url for the params is: {self._params}")
```

Comment:
Since we add this dictionary to the logs on line 92, maybe we don't need to display it on this line too. What do you think about that?